src/core/qgsvectorlayerlabeling.cpp

```diff
@@ -88,3 +92,454 @@ QgsVectorLayerSimpleLabeling *QgsVectorLayerSimpleLabeling::create( const QDomEl
 
   return new QgsVectorLayerSimpleLabeling( QgsPalLayerSettings() );
 }
+
+QPointF quadOffsetToSLDAnchor( QgsPalLayerSettings::QuadrantPosition quadrantPosition )
+{
+  double quadOffsetX = 0.5, quadOffsetY = 0.5;
+
+  // adjust quadrant offset of labels
+  switch ( quadrantPosition )
+  {
+    case QgsPalLayerSettings::QuadrantAboveLeft:
+      quadOffsetX = 1;
+      quadOffsetY = 0;
+      break;
+    case QgsPalLayerSettings::QuadrantAbove:
+      quadOffsetX = 0.5;
+      quadOffsetY = 0;
+      break;
+    case QgsPalLayerSettings::QuadrantAboveRight:
+      quadOffsetX = 0;
+      quadOffsetY = 0;
+      break;
+    case QgsPalLayerSettings::QuadrantLeft:
+      quadOffsetX = 1;
+      quadOffsetY = 0.5;
+      break;
+    case QgsPalLayerSettings::QuadrantRight:
+      quadOffsetX = 0;
+      quadOffsetY = 0.5;
+      break;
+    case QgsPalLayerSettings::QuadrantBelowLeft:
+      quadOffsetX = 1;
+      quadOffsetY = 1;
+      break;
+    case QgsPalLayerSettings::QuadrantBelow:
+      quadOffsetX = 0.5;
+      quadOffsetY = 1;
+      break;
+    case QgsPalLayerSettings::QuadrantBelowRight:
+      quadOffsetX = 0;
+      quadOffsetY = 1.0;
+      break;
+    case QgsPalLayerSettings::QuadrantOver:
+    default:
+      break;
+  }
+
+  return QPointF( quadOffsetX, quadOffsetY );
+}
+
+QgsUnitTypes::RenderUnit toRenderUnits( QgsPalLayerSettings::SizeUnit palUnit )
```

Comment:
Apologies, trying to understand... what is redundant? (and what's making it so?)